### DISCUSSION TITLE
Frequency form of song in editor

### DISCUSSION THREAD

**Discussion by astro-angelfish** [ppy/osu#29699] [Ideas]
When mapping some songs with instruments or vocals includes complex rhythms (e.g. hiphop/rap musics), it is hard to map the hit objects with correct timing with wave-form view as we can easily get confused with the sound before or after the sound we want to map.

I'm think if it is better if we could have a frequency form view, which might be helpful for distinguishing the sound to be mapped from the whole music?

**Comment by bdach** [comment]
You want a spectrogram display? How is that gonna help?

**Reply by astro-angelfish** [reply]
Yes but I did't know its exact name though.

Well I think the strength of sounds is transforming over time, and stronger sound might just cover the weaker sound with different frequency.

I want to map on the weaker sound which is covered by stronger sound, hard to distinguish in the wave form. I guess it would help in visual. This is an example I took from a [hiphop map](https://osu.ppy.sh/beatmapsets/1853196#mania/3988052) and broke the song down in audacity:
![image](https://github.com/user-attachments/assets/28312c64-127e-4e14-8b1d-2e5d96e00cd5)

Note I just guess this song contains 1/7, 1/9 rhythms. Maybe another proposal for these support needs a new thread.

But I also have to admit it will be used rarely because most song are pretty clear. It will only be useful when the mixing of the song is catastrophe, or the singer have problems in singing clearly.

By the way I'm running osu!lazer under linux so I'm just suffering from the extreme latency. So I had to map by visual.

**Reply by bdach** [reply]
A proposal for this was rejected once (https://github.com/ppy/osu/issues/8034) but it was a while ago. Not sure.
